README.md

```diff
@@ -29,9 +29,194 @@ This library is written in modern JavaScript and is published in both CommonJS a
+
+const keyPairFromMnemonic = await getKeyPairFromMnemonic(keyPair.mnemonic, keyPair.algorithm);
+// => Generates the same key pair based on the mnemonic.
+
+const keyPairFromSeed = await getKeyPairFromSeed(keyPair.seed, keyPair.algorithm);
+// => Generates the same key pair based on the seed.
+
+```
+
+> ⚠️ human-crypto-keys depends on cryptographical modules that can increase the size of a project. Please keep in mind that, depending on your needs, dynamic imports might be suitable.
+
+## API
+
+### generateKeyPair(algorithm, options)
+
+Generates a key pair based on the specified algorithm.
+
+Returns an object with the following:
+```js
+{
+    algorithm,  // An object with the algorithm identifier and respective parameters that were used during generation.
+    mnemonic,   // The mnemonic used to create a seed for generation.
+    seed,       // The seed used for generation.
+    privateKey, // The generated private key composed in a specific format.
+    publicKey,  // The generated public key composed in a specific format. 
+}
+```
+
+#### Parameters
+
+[algorithm](#algorithm)
+
+[options](#options)
+
+### getKeyPairFromMnemonic(mnemonic, algorithm, options)
+
+Generates a key pair based on the specified mnemonic and algorithm.
+
+Returns an object with the following:
+```js
+{
+    privateKey, // The generated private key composed in a specific format.
+    publicKey,  // The generated public key composed in a specific format. 
+}
+```
+
+#### Parameters
+
+<details><summary><strong>mnemonic</strong></summary>
+
+Type: `String`
+
+The mnemonic provided as one of the recovery methods for a key pair.
+</details>
+
+[algorithm](#algorithm)
+
+[options](#options)
+
+### getKeyPairFromSeed(seed, algorithm, options)
+
+Generates a key pair based on the specified seed and algorithm.
+
+Returns an object with the following:
+```js
+{
+    privateKey, // The generated private key composed in a specific format.
+    publicKey,  // The generated public key composed in a specific format. 
+}
 ```
 
+#### Parameters
+
+<details><summary><strong>seed</strong></summary>
+
+Type: `String`
+
+The seed provided as one of the recovery methods for a key pair.
+</details>
+
+[algorithm](#algorithm)
+
+[options](#options)
+
+### Common Parameters
+<details name="algorithm"><summary><strong>algorithm</strong></summary>
+ 
+Type: `Object` or `String`
+
+The algorithm identifier and the respective parameters to generate a key pair.
+
+It can be specified as an `Object` or a `String`. Using an `Object` will provide freedom to override default algorithm parameters in relation to its type. On the other hand, a `String` presents a useful and quick approach if the default parameters are suitable.
+
+The default parameters are different for each algorithm type. Currently only 2 types are supported:
+
+<details><summary><strong>RSA</strong></summary>
+
+Default Parameters:
+```js
+modulusLength: 2048		    // Number
```

Comment:
I think we should make this an object, similar to what we are doing below. The reason is that is not indented correctly in the README visualization itself.